File: python/passcraft.py
#!/usr/bin/env python3
from passcraft.password_generator import generate_password
from passcraft.passphrase_generator import generate_passphrase
import os
import sys

def load_word_list():
    """Load the word list from the text file"""
    word_list_path = os.path.join(os.path.dirname(__file__), '../enpass-word-list.txt')
    try:
        with open(word_list_path, 'r') as f:
            return [word.strip() for word in f.readlines() if word.strip()]
    except FileNotFoundError:
        print(f"Error: Word list file not found at {word_list_path}")
        sys.exit(1)

def main():
    word_list = load_word_list()
    print(f"Loaded {len(word_list)} words")
    
    while True:
        print("\n=== PassCraft Generator ===")
        print("1. Generate Password")
        print("2. Generate Passphrase")
        print("3. Exit")
        
        choice = input("Choose an option (1-3): ").strip()
        
        if choice == '1':
            generate_password()
        elif choice == '2':
            generate_passphrase(word_list)
        elif choice == '3':
            print("Goodbye!")
            sys.exit(0)
        else:
            print("Invalid choice. Please try again.")

if __name__ == "__main__":
    main()

File: java/src/PassCraft.java
package passcraft;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.List;
import java.util.Scanner;

public class PassCraft {
    private static List<String> wordList;
    private static final Scanner scanner = new Scanner(System.in);

    public static void main(String[] args) {
        try {
            // Load word list
            wordList = Files.readAllLines(Paths.get("../enpass-word-list.txt"));
            System.out.println("Loaded " + wordList.size() + " words");
            
            while (true) {
                showMainMenu();
            }
        } catch (IOException e) {
            System.err.println("Error loading word list: " + e.getMessage());
            System.exit(1);
        }
    }

    private static void showMainMenu() {
        System.out.println("\n=== PassCraft Generator ===");
        System.out.println("1. Generate Password");
        System.out.println("2. Generate Passphrase");
        System.out.println("3. Exit");
        System.out.print("Choose an option (1-3): ");

        int choice = getIntInput(1, 3);

        switch (choice) {
            case 1:
                PasswordGenerator.generate();
                break;
            case 2:
                PassphraseGenerator.generate(wordList);
                break;
            case 3:
                System.out.println("Goodbye!");
                System.exit(0);
        }
    }

    public static int getIntInput(int min, int max) {
        while (true) {
            try {
                int input = Integer.parseInt(scanner.nextLine());
                if (input >= min && input <= max) {
                    return input;
                }
                System.out.print("Invalid input. Please enter a number between " + min + "-" + max + ": ");
            } catch (NumberFormatException e) {
                System.out.print("Invalid input. Please enter a number: ");
            }
        }
    }
}